source/Octopus.Client/Repositories/Async/MixedScopeBaseRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Extensibility;
+using Octopus.Client.Util;
+
+namespace Octopus.Client.Repositories.Async
+{
+    class MixedScopeBaseRepository<TMixedScopeResource>: BasicRepository<TMixedScopeResource> where TMixedScopeResource : class, IResource
+    {
+        public MixedScopeBaseRepository(IOctopusAsyncClient client, string collectionLinkName) : base(client, collectionLinkName)
+        {
+        }
+
+        SpaceQueryParameters SpaceQueryParameters { get; set; }
+        protected void SetupParameters(bool includeGlobal, string[] spaceIds)
+        {
+            ParameterHelper.ValidateSpaceParameters(SpaceQueryParameters, includeGlobal, spaceIds);
+            SpaceQueryParameters = new SpaceQueryParameters(){IncludeGlobal = includeGlobal, SpaceIds = spaceIds};
+        }
+
+        protected override Dictionary<string, object> AdditionalQueryParameters
+        {
+            get
+            {
+                if (SpaceQueryParameters == null)
+                    return null;
```

Comment:
We can return an empty dictionary here and make the contract of `AdditionalQueryParameters` a little more strict in that it can't be null. We control all implementations so this should be safe.